src/include/threadpool/worker_pool.h

```diff
@@ -20,9 +20,16 @@
 namespace peloton {
 namespace threadpool {
 
+class WorkerPool;
+
 using TaskQueue = peloton::LockFreeQueue<std::function<void()>>;
 
-void WorkerFunc(std::atomic_bool *should_shutdown, TaskQueue *task_queue);
+/**
+ * Wrapper method for invoking a function inside of the worker pool
+ * @param should_shutdown A pointer to boolean that tells whether to shutdown
+ * @param task_queue Where to poll for tasks
+ */
+void WorkerFunc(WorkerPool *pool, TaskQueue *task_queue);
```

Comment:
Since this function isn't needed in the header, you can move it to the cpp file in an anonymous namespace.